extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/client/HashicorpVaultClient.java

```diff
@@ -276,11 +276,17 @@ private HttpUrl getSecretUrl(String key, String entryType) {
         key = key.replace("%2F", "/");
 
         var vaultApiPath = settings.secretPath();
+        var folderPath = settings.getFolderPath();
+
+        if (folderPath == null) {
+            folderPath = "";
+        }
```

Comment:
in case of null folder path it would be better to avoid to add a path segments at all because:
- it will be more explicit
- it will avoid unnecessary calls to the `trimLeadingOrEndingSlash` method 
- it will not rely on an internal detail of okhttp (the fact that an empty string is ignored.)

Reply:
in case of null, the `folderPath` will not be added now